### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by mar-ses** [dfm/python-bls#4]
Hey, I keep getting this "Segmentation fault (core dumped)" when I try to run the BLS on a lightcurve, and I've never used Fortran before so I'm not sure what it means. I attached an example lightcurve I was trying to fit as a csv, and here are the arguments I was using for `eebls`:

```
u = np.empty(len(lcf))         # Is this correct, what is the required dtype for the u, v work arrays?
v = np.empty(len(lcf))

nf = 500
fmin = 1/8.
fmax = 39.443686562975699
df = int((fmax-fmin)//nf)
qmi = 0.5/24
qma = 1.

nb = 1772

power, best_period, best_power, depth, _, _, _ = bls.eebls(lcf['t'], lcf['f_detrended'], u, v, nf, fmin, df, nb, qmi, qma)
```

Also, for nb I'm not sure what to put, since apparently the max is 2000. Does nb have to be a factor of the number of cadences or lightcurve points? In this case, I cut out one cadence from lcf to make it even i.e: 

`lcf = lcf.loc[:(nb*2-1)]`

so that 

`len(lcf) = 3544 = 1772 * 2`

and same with `len(u) = 3544` and `v`.

lightcurve:
[lc_example.txt](https://github.com/dfm/python-bls/files/896196/lc_example.txt)

**Comment by dfm** [comment]
Have you ever successfully run this code in the past? Can you narrow down what setting of the parameters is causing the problem? We aren't really thoroughly supporting this package anymore and I haven't touched it in many years but I'm happy to help and try to track down the problem.

**Comment by mar-ses** [comment]
Afterwards it started running successfully, but only when using `nb ~< 2000`. However, in the example code posted above, some of the arguments I was using are bad; I entered period instead of frequency and the qmin and qmax are also wrong. 

As it seems to work unless nb is a certain value, I don't mind if it gets fixed or not so no worries, I'm happy to close the issue. 

Just in case you were interested though, I found that for the above lightcurve, when `nb <= 1671` there is no error, but when `nb > 1671` there is. It's probably something to do with the arguments I'm inputting, but I can't see what. Here are the argument values that work normally otherwise:

```
(Pdb) qmi
0.010416666666666666
(Pdb) qma
0.2
(Pdb) df
0.00098732370010085882
(Pdb) fmin
0.025352599798282097
(Pdb) nf
2000
(Pdb) len(lcf)            # number of points in lightcurve
3545
```
